components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```
@@ -8693,7 +8776,20 @@ public void deleteSharedScope(String scopeName, String tenantDomain) throws APIM
     @Override
     public void updateSharedScope(Scope sharedScope, String tenantDomain) throws APIManagementException {
 
-        KeyManagerHolder.getKeyManagerInstance(tenantDomain).updateScope(sharedScope);
+        int tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+        Map<String, KeyManagerDto> tenantKeyManagers = KeyManagerHolder.getTenantKeyManagers(tenantDomain);
+        for (Map.Entry<String, KeyManagerDto> keyManagerEntry : tenantKeyManagers.entrySet()) {
+            KeyManager keyManager = keyManagerEntry.getValue().getKeyManager();
+            if (keyManager != null) {
+                try{
+                    keyManager.updateScope(sharedScope);
+                }catch (APIManagementException e){
```

same here. formatting issues.
Shouldnt we throw the exception?